control_panel_api/views.py

```diff
@@ -58,6 +58,15 @@ class AppS3BucketViewSet(viewsets.ModelViewSet):
     queryset = AppS3Bucket.objects.all()
     serializer_class = AppS3BucketSerializer
 
+    def perform_create(self, serializer):
+        app_s3bucket = serializer.save()
+
+        services.app_granted_access_to_bucket(
```

Comment:
I think we should call the service method here and pass an instance, not many parameters, and encapsulate the readwrite logic inside the service layer - view stays clean - could look like this:

```
def perform_create(self, serializer):
    app_s3bucket = serializer.save()
    services.app_s3bucket_create(app_s3bucket)
```

Could also refactor existing view/service and use OO properly and just pass the instance and keep the views clean.

Reply:
I tried to do exactly that in another of the services function but had troubles with testing (it was on deletion of something so the `assert_called_with()` was confused by the missing id on the argument or something like that)...

...in the end I decided to keep that function consistent with the rest. Again this is more general question, why this function receive a model instance but most of the others receive `app_slug`, `bucket_name`, etc...? Should we change all of them? Is 3 an high number of arguments?

Reply:
I'll change it 👍

Reply:
FYI: Another thing to bear in mind here is that we're coupling the `services` layer with the models which is not strictly necessary.

e.g. see the tests of this function now, I'm just calling it passing some strings, now that I'm changing it to receive a model instance I'll have to create a `AppS3Bucket` record just to test the function to grant access to an app. This makes me wonder about the design of it.